include/ncengine/asset/AssetData.h

```diff
@@ -30,9 +30,10 @@ struct TextureWithId
 {
     asset::Texture texture;
     std::string id;
+    bool isNormalMap;
```

Comment:
Is passing the flags here better for potential changes in the future? I haven't looked at all the code yet, but on first look, it seems easier to pass the whole set of flags to whatever is actually acting on them.